src/lib/steps.ts

```diff
@@ -144,6 +144,13 @@ export type SortStep = {
   columns: SortColumnType[];
 };
 
+export type SubstringStep = {
+  name: 'substring';
+  column: string;
+  start_index: number;
+  number_characters: number;
```

Comment:
I would use `length` instead of `number_characters`. Besides, I'm wondering if we shouldn't use the same semantic as in python and use `start_index`, `stop_index`, make `stop_index` optional and use `$strLenCP` to compute actual stop index if `stop_index` is negative.

What do you think?